src/Badges.sol

```diff
@@ -35,8 +39,20 @@ contract Badges is
 
   mapping(uint256 => uint256) private voucherHashIds;
 
+  mapping(uint256 => bool) public revokedBadges;
+  mapping(uint256 => uint256) public badgeExpirationDates;
   event SpecCreated(address indexed to, string specUri, uint256 indexed raftTokenId, address indexed raftAddress);
 
+  modifier senderIsRaftOwner(uint256 _raftTokenId) {
```

Comment:
Maybe best to pass is the message string as a parameter so we know which function's context is propagated from. With this, all errors will show as "senderIsRaftOwner: unauthorized" instead of "createSpec: unauthorized".

Wdyt?

Reply:
yep, good idea. It felt kind of off to lose that context